### ISSUE TITLE
Test Smell: it is not a good test practice to use the loop in the test

### ISSUE THREAD

**Issue by TestSmell** [biojava/biojava#1038]
Hi!

We notice that you use the loop structure in your test cases.
For example, testQualityScoresIntArray() in FastqToolsTest.java
<img width="497" alt="截屏2022-08-17 下午5 17 26" src="https://user-images.githubusercontent.com/110402847/185083359-94544a0a-e17c-454f-9406-d4fa380e6774.png">

However, using the loop in test cases is not a good test practice.
We analyzed the relevant Stack Overflow posts and summarized four potential negatives it brings:
1. Loops make the test case more complex
2. In most cases, a loop can be replaced with a data-driven test that is more readable.
3. Loops break the assert-for-one-thing thumb rule. I don't mean a single assert statement.
4. When a test fails, knowing the reason is more complicated.

Solution:
To avoid using the loop in the test, JUnit provides an annotation (i.e., @ParameteredTest), enabling a test case to run multiple times with different parameters.
We provide a usage example here:
<img width="622" alt="image" src="https://user-images.githubusercontent.com/110402847/185083811-1d731ae3-0188-475f-b378-714beb492e7a.png">

**Comment by josemduarte** [comment]
Good point, thanks! Would you be able to provide a pull request?

**Comment by tmrpavan** [comment]
@josemduarte I would like to work on this. We are using  Junit4 is there any chance to migrate Junit5. So this type of test case are easy to write in Junit5

**Comment by josemduarte** [comment]
Yes we have in fact converted some tests to junit5 and the agreement is that new tests should be junit5.

**Comment by tmrpavan** [comment]
can you assign this issue to me. I will work on this

**Comment by arkanar611** [comment]
I've created two merge requests for the genome module:
this one https://github.com/biojava/biojava/pull/1125  includes migration to junit5 only for the FastqToolsTest.java
And this one https://github.com/biojava/biojava/pull/1126 migrates all genome module tests to junit5

MRs can be reviewed separately or we can merge only the last one because it includes all changes
